Content.Server/GameObjects/Components/Weapon/Ranged/Projectile/AmmoWeapon.cs

```diff
@@ -0,0 +1,145 @@
+
+        protected BallisticCaliber Caliber ;
+        /// <summary>
+        ///     What type of ammo this gun can fire.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public BallisticCaliber GunCaliber { get => Caliber; set => Caliber = value; }
+
+        private string _soundGunEmpty;
+        /// <summary>
+        ///     Sound played when trying to shoot if there is no ammo available.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public string SoundGunEmpty { get => _soundGunEmpty; set => _soundGunEmpty = value; }
+
+        private float _spreadStdDev_Gun;
+        /// <summary>
+        ///     Increases the standard deviation of the ammo being fired.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float SpreadStdDev_Gun { get => _spreadStdDev_Gun; set => _spreadStdDev_Gun = value; }
+
+        private float _evenSpreadAngle_Gun;
+        /// <summary>
+        ///     Increases the evenspread of the ammo being fired.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float EvenSpreadAngle_Gun { get => _evenSpreadAngle_Gun; set => _evenSpreadAngle_Gun = value; }
+
+        private float _velocity_Gun;
+        /// <summary>
+        ///     Increases the velocity of the ammo being fired.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float Velocity_Gun { get => _velocity_Gun; set => _velocity_Gun = value; }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _soundGunEmpty, "sound_empty", "/Audio/Guns/Empty/empty.ogg");
+            serializer.DataField(ref _spreadStdDev_Gun, "spreadstddev", 0);
+            serializer.DataField(ref _evenSpreadAngle_Gun, "evenspread", 0);
+            serializer.DataField(ref _velocity_Gun, "gunvelocity", 0);
+            serializer.DataField(ref Caliber, "caliber", BallisticCaliber.Unspecified);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            Owner.GetComponent<RangedWeaponComponent>().FireHandler = TryShoot;
+            Chambers = new Chamber[ChamberCount];
+            for (var i = 0; i < Chambers.Length; i++)
+            {
+                var container = ContainerManagerComponent.Ensure<ContainerSlot>($"ballistics_chamber_{i}", Owner);
+                Chambers[i] = new Chamber(container);
+            }
+        }
+
+        /// <summary>
+        ///     Fires projectiles based on loaded ammo from entity to a coordinate.
+        /// </summary>
+        protected void TryShoot(IEntity user, GridCoordinates clickLocation)
+        {
+            var ammo = GetChambered(FirstChamber)?.GetComponent<BallisticBulletComponent>();
+            CycleChamberedBullet(FirstChamber);
+            if (ammo == null | ammo?.Spent == true | ammo?.Caliber != Caliber)
```

Comment:
You're using the bitwise or operator here instead of the logical one. Is that intentional?